dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -192,6 +200,153 @@ void appendWindowDescription(
     window_columns.emplace_back(func_string, result_type);
     source_columns.emplace_back(func_string, result_type);
 }
+
+void setAuxiliaryColumnInfoImpl(
+    WindowDescription & window_desc,
+    const String & aux_col_name,
+    const Block & tmp_block,
+    bool is_begin)
+{
+    if (!aux_col_name.empty())
+    {
+        // Prepare something
+        Int32 & range_auxiliary_column_index = is_begin ? window_desc.frame.begin_range_auxiliary_column_index : window_desc.frame.end_range_auxiliary_column_index;
+        TypeIndex & aux_col_type = is_begin ? window_desc.begin_aux_col_type : window_desc.end_aux_col_type;
+
+        // Set auxiliary columns' indexes
+        size_t aux_col_idx = tmp_block.getPositionByName(aux_col_name);
+        range_auxiliary_column_index = aux_col_idx;
+
+        // Set auxiliary columns' types
+        const auto & col_and_name = tmp_block.getByName(aux_col_name);
+        auto data_type = col_and_name.type;
+        if (data_type->isNullable())
+            throw Exception("Get an unexpected nullable column when the frame type is range");
+        aux_col_type = data_type->getTypeId();
+    }
+}
+
+// We need auxiliary columns' info when finding the start or end boundary of the frame
+void setAuxiliaryColumnInfo(
+    ExpressionActionsPtr & actions,
+    WindowDescription & window_desc,
+    const String & begin_aux_col_name,
+    const String & end_aux_col_name,
+    const tipb::Window & window)
+{
+    // Execute this function only when the frame type is Range
+    if (window.frame().type() != tipb::WindowFrameType::Ranges)
+        return;
+
+    if (begin_aux_col_name.empty() && end_aux_col_name.empty())
+        return;
+
+    const Block & tmp_block = actions->getSampleBlock();
+    setAuxiliaryColumnInfoImpl(window_desc, begin_aux_col_name, tmp_block, true);
+    setAuxiliaryColumnInfoImpl(window_desc, end_aux_col_name, tmp_block, false);
+}
+
+void setOrderByColumnTypeAndDirection(WindowDescription & window_desc, const ExpressionActionsPtr & actions, const tipb::Window & window)
+{
+    // Execute this function only when the frame type is Range
```

Comment:
why?
Can some explaining notes be added?
And it looks like the function name could be changed to `setOrderByColumnTypeAndDirectionForRangeFrame`